lib/structures/Message.js

```diff
@@ -68,6 +68,9 @@ class Message extends Base {
         if(data.author) {
             if(data.author.discriminator !== "0000") {
                 this.author = client.users.add(data.author, client);
+                if(this.author!==data.author){
+                    this.author.update(data.author);
+                }
```

Comment:
This as a whole can be merged together into `this.author = client.users.update(...)` on line 70

Reply:
no we don't want to perform an update in the collection item. It would lead to invalid reference in other places.
We just want to update the Object itself. We do so by calling `<user>.update`

Reply:
edit: I'm just dumb. the whole thing (eventhe first .add) can indeed be replaced by `client.users.update(data.author, client)`